ls1intum/Artemis
# `Plagiarism checks`: Fix memory issues with large comparisons and add feature toggle

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


The plagiarism checks can be heavyweight and cause serious issues on the server. Also, loading the plagiarism elements by the native query might cause several issues.

### Description

This PR adds a feature toggle to disable plagiarism checks, e.g. during exams, as they can be quite heavyweight.
It also fixed some warnings in TextExerciseResource and ModelingExerciseResource and modernizes/simplifes their code with respect to authorization checks

### Steps for Testing

Prerequisites:

1 Instructor
2 Students
1 Exercise with two very similar submissions by both students (e.g. text exercise with lorem ipsum)

1a. Login to Artemis as Instructor
1b. Navigate to the exercise detail page
1c. Choose check plagiarism
1d. Detect plagiarism cases
1e. Confirm the plagiarism case

2a. Navigate to Course Detail
2b. Open Plagiarism Cases
2c. Asses the plagiarism case and save instructor statement

2a. Log in to Artemis as Student
2b. Open Notifications
2c. Click on the Plagiarism Notification
2d. Respond to the plagiarism Case

3a. Login in again to Artemis as Instructor
3b. Navigate to Course Detail
3c. Open Plagiarism Cases
3d. Assess the student's response

4a. Log in to Artemis again as Student
4b. Open Notifications
4c. Click on the Plagiarism Notification
4d. Check that the status has been updated by the instructor

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots
